mllib/src/main/scala/org/apache/spark/mllib/linalg/Vectors.scala

```diff
@@ -717,6 +719,23 @@ class SparseVector(
       new SparseVector(size, ii, vv)
     }
   }
+
+  override def argmax: Int = {
+    if (size == 0) {
+      -1
+    } else {
+      var maxIdx = indices(0)
+      var maxValue = values(0)
+
+      foreachActive { (i, v) =>
```

Comment:
The bug still exists. Take a sparse vector `val sv = SparseVector(5, Array(1, 3), Array(-1, -5))` for example, `sv.argmax` should return one from `{0, 2, 4}` because `0` is the max value in this vector.